### PR TITLE
Fixes errors with making temp rank changes before permanent ones

### PR DESCRIPTION
**Pull request by Jordie0608**
Also makes permission panel grouped by ranks sorted alphabetically.

### CODE REVIEW

File: code/modules/admin/permissionedit.dm [hunk 1]
```diff
@@ -130,11 +130,14 @@
...
-	. = ckey(input("New admin's ckey","Admin ckey") as text|null)
+/datum/admins/proc/add_admin(admin_ckey, use_db)
+	if(admin_ckey)
+		. = admin_ckey
+	else
+		. = ckey(input("New admin's ckey","Admin ckey") as text|null)
 	if(!.)
 		return FALSE
-	if(. in GLOB.admin_datums+GLOB.deadmins)
+	if(!admin_ckey && . in GLOB.admin_datums+GLOB.deadmins)
```
Comment (Cyberboss) [inline code/modules/admin/permissionedit.dm]: Doesn't do what you think it does, add parenthesis

File: code/modules/admin/permissionedit.dm [hunk 2]
```diff
@@ -195,19 +198,27 @@
 		GLOB.admin_ranks += R
 	if(use_db)
 		new_rank = sanitizeSQL(new_rank)
-		if(!R)
-			var/datum/DBQuery/query_add_rank = SSdbcore.NewQuery("INSERT INTO [format_table_name("admin_ranks")] (rank, flags, exclude_flags, can_edit_rights) VALUES ('[new_rank]', '0', '0', '0')")
+		//if a player was tempminned before having a permanent change made to their rank they won't yet be in the db
+		var/old_rank
+		var/datum/DBQuery/query_admin_in_db = SSdbcore.NewQuery("SELECT rank FROM [format_table_name("admin")] WHERE ckey = '[admin_ckey]'")
```
Comment (Cyberboss) [inline code/modules/admin/permissionedit.dm]: sanitize you madlad!

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/admin/permissionedit.dm (hunk 2 above)
Comment (Jordie0608) [reply code/modules/admin/permissionedit.dm]: This value is already sanitized before being passed down.

Comment (Cyberboss) [reply code/modules/admin/permissionedit.dm]: pretty sure it'd be trivial for kevinz to find a way to procall this though

Comment (Cyberboss) [reply code/modules/admin/permissionedit.dm]: wait, the safteies, nvm

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**